cdap-data-fabric/src/main/java/io/cdap/cdap/store/NamespaceTable.java

```diff
@@ -105,4 +108,23 @@ public void delete(NamespaceId id) throws IOException {
     }
     return result;
   }
+
+  /**
+   * Count all namespaces
+   * @return long of all namespaces except for system.
+   * @throws IOException from StructuredTable.count
+   */
+  public long getNamespaceCount() throws IOException {
+    return getNamespaceCount(Collections.singletonList(Range.all()));
+  }
+
+  /**
+   * Count namespaces for a particular range.
+   * @param ranges list of ranges
+   * @return long of all namespaces based on range
+   * @throws IOException from StructuredTable.count
+   */
+  public long getNamespaceCount(Collection<Range> ranges) throws IOException {
```

Comment:
When this is used??

Reply:
By the above function, which passes in Range.all().

Reply:
Why need two functions then??

Reply:
In case in the future we need special ranges for namespace counting.